ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +277,151 @@ void PowerSystemNode::injectFault (const string& fault_name,
     {
       ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
       wattage += hpd_wattage;
-    } else
+    }
+  }
+}
+
+void PowerSystemNode::injectCustomFault(bool& fault_activated,
+                                        const PrognoserVector& sequence,
+                                        size_t& index,
+                                        double& wattage,
+                                        double& voltage,
+                                        double& temperature)
+{
+  static string saved_fault_directory = "N/A";
+  string current_fault_directory;
+  string designated_file;
+  static string saved_file;
+  static bool custom_fault_ready = false;
+  static bool custom_warning_displayed = false;
+  static bool end_fault_warning_displayed = false;
+  bool fault_enabled = false;
+
+  // Do nothing unless the specified fault has been injected.
+  if (! ros::param::getCached("/faults/activate_custom_fault", fault_enabled)) {
+    return;
+  }
+
+  if (!fault_activated && fault_enabled)
+  {
+    // Multiple potential points of failure, so alert user the process has started.
+    ROS_INFO_STREAM("Attempting custom fault activation...");
+
+    // Get user-entered file directory.
+    if (! ros::param::getCached("/faults/custom_fault_profile", designated_file))
+    {
+      ROS_ERROR("Failed to fetch custom fault profile name.");
+      return;
+    }
+
+    // Append the current fault directory to the stored file path.
+    current_fault_directory = ros::package::getPath("ow_power_system") + "/profiles/" + designated_file;
+    
+    if (current_fault_directory != saved_fault_directory || end_fault_warning_displayed)
     {
-      // DEPRECATED (7/12/22)
-      // This code was used to inject the values from the CSV-defined power faults
-      // when they are activated. The CSV fault system is not currently used by
-      // the high power draw fault, but it will be revisited in the coming days/weeks, so
-      // it has not been fully removed yet.
-
-      /*
-      // TODO: Unspecified how to handle end of fault profile, which is
-      // unlikely.  For now, reuse the last entry.
-      if (index + m_profile_increment >= sequence.size()) {
-        ROS_WARN_STREAM_ONCE
-          (fault_name << ": reached end of fault profile, reusing last entry.");
-        // Probably unneeded, but makes index explicit.
-        index = sequence.size() - 1;
+      // It's a different fault profile from whatever the previous stored directory was, OR the
+      // previous fault ran its course, so reset index.      
+      // Load a new fault profile from the provided directory.
+      try
+      {
+        ifstream file(current_fault_directory);
+        if (current_fault_directory.substr(current_fault_directory.size() - 4) != ".csv")
+        {
+          throw ERR_INVALID_FILE_NAME;
+        }
+        else if (file.fail())
+        {
+          throw ERR_NO_CSV_FOUND;
+        }
```

Comment:
This ifstream is only created here for the sole purpose of interrogating whether the file exists, but creating an ifstream and checking its existence is already performed by `PowerSystemNode::loadPowerProfile`, which is called later in the try block. 

I suggesting removing these lines and the ifstream above.